File: languages/php/Bloock/Entity/Authenticity/SignatureAlg.php
<?php

namespace Bloock\Entity\Authenticity;

use Bloock\SignerAlg;

/**
 * Represents different signature algorithms.
 */
class SignatureAlg
{
    /**
     * Represents the ECDSA signature algorithm with the "ES256K" name.
     */
    const ECDSA = "ECDSA";
    /**
     * Represents the ENS (Ethereum Name Service) signature algorithm.
     */
    const ENS = "ENS";
    /**
     * Represents the BJJ signature algorithm with the "BJJ" name.
     */
    const BJJ = "BJJ";
    /**
     * Represents an unrecognized signature algorithm.
     */
    const UNRECOGNIZED = "UNRECOGNIZED";

    /**
     * Converts a string representation of an algorithm to the corresponding SignatureAlg enum.
     * @param string $alg
     * @return string
     */
    public static function fromString(string $alg): string
    {
        switch ($alg) {
            case "ES256K":
                return SignatureAlg::ECDSA;
            case "ENS":
                return SignatureAlg::ENS;
            case "BJJ":
                return SignatureAlg::BJJ;
            default:
                return SignatureAlg::UNRECOGNIZED;
        }
    }

    public static function fromProto(SignerAlg $alg): string
    {
        switch ($alg) {
            case SignerAlg::ES256K:
                return SignatureAlg::ECDSA;
            case SignerAlg::ENS:
                return SignatureAlg::ENS;
            case SignerAlg::BJJ:
                return SignatureAlg::BJJ;
            default:
                return SignatureAlg::UNRECOGNIZED;
        }
    }
}

File: languages/python/bloock/entity/authenticity/signature_alg.py
from __future__ import annotations

from enum import Enum


class SignatureAlg(Enum):
    """
    Represents different signature algorithms.
    """
    ECDSA = 0
    """
    Represents the ECDSA signature algorithm with the "ES256K" name.
    """
    ENS = 1
    """
    Represents the ENS (Ethereum Name Service) signature algorithm.
    """
    BJJ = 2
    """
    Represents the BJJ signature algorithm with the "BJJ" name.
    """
    UNRECOGNIZED = -1
    """
    Represents an unrecognized signature algorithm.
    """

    def __int__(self):
        return self.value

    @staticmethod
    def from_str(alg: str) -> SignatureAlg:
        """
        Converts a string representation of an algorithm to the corresponding SignatureAlg enum.
        :type alg: object
        :rtype: object
        """
        if alg == "ES256K":
            return SignatureAlg.ECDSA
        elif alg == "ENS":
            return SignatureAlg.ENS
        elif alg == "BJJ":
            return SignatureAlg.BJJ
        else:
            return SignatureAlg.UNRECOGNIZED
